pkg/apis/configuration/v1alpha1/types.go

```diff
@@ -36,31 +36,31 @@ type Configuration struct {
 // ConfigurationSpec is the spec for an configuration.
 type ConfigurationSpec struct {
 	// +optional
-	AppHTTPPipelineSpec PipelineSpec `json:"appHttpPipeline,omitempty"`
+	AppHTTPPipelineSpec *PipelineSpec `json:"appHttpPipeline,omitempty"`
```

Comment:
They were needed in #6101. When I made the change to drop K8s support, I left these here as pointers because they had `omitempty` and in this case it's not a bad thing to make them pointers anyways. Should we re-add K8s support in the future, this will help.